core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -1171,6 +1171,7 @@ private[spark] object JsonProtocolSuite extends Assertions {
       |          "Replication": 1
       |        },
       |        "Barrier" : false,
+      |        "DeterministicLevel" : "DETERMINATE",
```

Comment:
It might be good to add an INDETERMINATE test case to this file.

Reply:
Okay. let me add a test case for it.